### DISCUSSION TITLE
rasa-webchat "The client is using an unsupported version of the Socket.IO or Engine.IO protocols"

### DISCUSSION THREAD

**Discussion by ptrkdy** [botfront/botfront#979] [Q&A]
Hi team,

I am using botfront 1.0.4, things are pretty cool! Great project :)

I've got almost everything working, but have spent the last day or so trying to troubleshoot the chat widget, so thought I would start a discussion here.

When attempting to use either the latest botfront with the install / embed webchat widget, or alternatively using the rasa-webchat widget, I consistently encounter the error "The client is using an unsupported version of the Socket.IO or Engine.IO protocols" when embedding on my website.

status code: 400
response: "The client is using an unsupported version of the Socket.IO or Engine.IO protocols"
Browser: Chrome v88 & Firefox 84.0.2

Any assistance would be greatly appreciated.

Thanks!
Patrick

**Comment by ptrkdy** [comment]
Adding some more context:

I've tried to version pin @0.11.12 & @1.0.0 version of rasa-webchat javascript, available at the following CDN:

https://cdn.jsdelivr.net/npm/rasa-webchat@0.11.12/
https://cdn.jsdelivr.net/npm/rasa-webchat@1.0.0/

I've also reviewed the PR that updated the version to 1.0.0, and tried the legacy version of the script (with 0.11.12) as well as new version in the current repo (1.0.0) and both are providing the same response code: "The client is using an unsupported version of the Socket.IO or Engine.IO protocols"

https://github.com/botfront/rasa-webchat/commit/bf3519361c03ae195fdf01cf773630ad288776a9

In other words, I have tested both of the following blocks of code:

`<div id="webchat"></div>`

`<script src="https://cdn.jsdelivr.net/npm/rasa-webchat@0.11.12/lib/index.min.js"></script>`

`<script>
  WebChat.default.init({
    selector: "#webchat",
    initPayload: "/get_started",
    customData: {"language": "en"}, // arbitrary custom data. Stay minimal as this will be added to the socket
    socketUrl: "https://socket.myworkingsocketdomain.com",
    socketPath: "/socket.io/",
    title: "Title",
    subtitle: "Subtitle",
    params: {"storage": "session"} // can be set to "local"  or "session". details in storage section.
  })
</script>`

annnnnnd:

`<script>!(function () {
  let e = document.createElement("script"),
    t = document.head || document.getElementsByTagName("head")[0];
  (e.src =
    "https://cdn.jsdelivr.net/npm/rasa-webchat@1.0.0/lib/index.js"),
    (e.async = !0),
    (e.onload = () => {
      window.Webchat.default(
        {
          customData: { language: "en" },
          socketUrl: "https://socket.myworkingsocketdomain.com",
          // add other props here
        },
        null
      );
    }),
    t.insertBefore(e, t.firstChild);
})();
</script>`

Once again, any further assistance would be greatly appreciated!

Thanks!
Patrick

**Comment by ptrkdy** [comment]
Also appears that the socket.io client version is available in package.json as ^2.2.0

https://github.com/botfront/rasa-webchat/commit/bf3519361c03ae195fdf01cf773630ad288776a9#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L38

https://github.com/botfront/rasa-webchat/blob/master/package.json#L38

I'm not sure how to determine how this is supported from server side in botfront project - I'm not familiar with the meteor framework.

Curious,
Patrick

**Comment by ptrkdy** [comment]
I think this version increase in socket.io client is in the following PR:

https://github.com/botfront/rasa-webchat/pull/346

Once merged and hosted on CDNs I think it will work.

Thanks

**Comment by MatthieuJnon** [comment]
yes, it's updated, make sure you select version 1.0.1.

Le jeu. 15 avr. 2021 à 17:31, Ayush Kumar ***@***.***> a
écrit :

> It's been merged but I think CDN is not updated
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/botfront/botfront/discussions/979#discussioncomment-617817>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGA7UGEJPAWPQNOJ54SEA2DTI5LKZANCNFSM4ZRFH73Q>
> .
>


-- 
*Matthieu Joannon*
Engineering Lead
botfront.io
